runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/source/bounded/FlinkBoundedSourceReader.java

```diff
@@ -61,8 +55,6 @@
  */
 public class FlinkBoundedSourceReader<T> extends FlinkSourceReaderBase<T, WindowedValue<T>> {
   private static final Logger LOG = LoggerFactory.getLogger(FlinkBoundedSourceReader.class);
-  private static final VarLongCoder LONG_CODER = VarLongCoder.of();
-  private final Map<Integer, Long> consumedFromSplit = new HashMap<>();
```

Comment:
Can you please elaborate how does removing tracking of consumption from state helps to fix the bug, to not reemit elements from impulse source?

Reply:
This is related, because this was introduce in an attempt to fix exactly this issue of avoiding re-emission of elements from Impulse. Unfortunately, this is not working as expected and - more seriously - can cause data-loss for sources that are not 'stable' - e.g. NoSQL databases, where the data in a split can change between restarts, which might cause skipping an element that was always present (but changed its ordered position).

Reply:
Understood. Thanks 👍